ext/opentelemetry-ext-grpc/src/opentelemetry/ext/grpc/client_interceptor.py

```diff
@@ -0,0 +1,217 @@
+# Copyright 2020, OpenTelemetry Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import collections
+
+import grpc
+from grpc.framework.foundation import future
+from grpc.framework.interfaces.face import face
+
+
+class _ClientCallDetails(
+    collections.namedtuple(
+        "_ClientCallDetails", ("method", "timeout", "metadata", "credentials")
+    ),
+    grpc.ClientCallDetails,
+):
+    pass
+
+
+# TODO
+# https://github.com/c24t/grpc/blob/50619a562f50ef7ee680ed5ce35de94b5a3b1539/src/python/grpcio/grpc/__init__.py#L561
+# https://github.com/census-instrumentation/opencensus-python/pull/617/files#diff-16ff5c7896222cfa69b7aed98860f7d3R50
+class WrappedResponseIterator(future.Future, face.Call):
+    """Wraps the rpc response iterator.
+
+    The grpc.StreamStreamClientInterceptor abstract class states stream
+    interceptor method should return an object that's both a call (implementing
+    the response iterator) and a future.  Thus, this class is a thin wrapper
+    around the rpc response to provide the opencensus extension.
+
+    :type iterator: (future.Future, face.Call)
+    :param iterator: rpc response iterator
+
+    :type span: opencensus.trace.Span
+    :param span: rpc span
+    """
+
+    def __init__(self, iterator, span):
+        self._iterator = iterator
+        self._span = span
+
+        # TODO
+        # self._messages_received = 0
+
+    def add_done_callback(self, fn):
+        self._iterator.add_done_callback(lambda ignored_callback: fn(self))
+
+    def __iter__(self):
+        return self
+
+    def __next__(self):
+        try:
+            message = next(self._iterator)
+        except StopIteration:
+            print("end wrapper iterator")
+            # execution_context.get_opencensus_tracer().end_span()
+            raise
+
+        # self._messages_received += 1
+        # add_message_event(
+        #     proto_message=message,
+        #     span=self._span,
+        #     message_event_type=time_event.Type.RECEIVED,
+        #     message_id=self._messages_received)
+
+        return message
+
+    def next(self):
+        return self.__next__()
+
+    def cancel(self):
+        return self._iterator.cancel()
+
+    def is_active(self):
+        return self._iterator.is_active()
+
+    def cancelled(self):
```

Comment:
Can all these `NotImplementedError` be handled by `__getattr__` to avoid repeated code?